components/org.wso2.carbon.identity.api.user.application/org.wso2.carbon.identity.rest.api.user.application.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/application/v1/core/ApplicationService.java

```diff
@@ -181,20 +186,22 @@ private Link buildPageLink(String rel, int offset, int limit) {
     private String buildFilter(String filter) {
 
         if (StringUtils.isNotBlank(filter)) {
-            String[] filterArgs = filter.split(" ");
-            if (filterArgs.length == 3) {
+            try {
+                FilterTreeBuilder filterTreeBuilder = new FilterTreeBuilder(filter);
+                Node rootNode = filterTreeBuilder.buildTree();
 
-                String filterAttribute = filterArgs[0];
+                if (!(rootNode instanceof ExpressionNode)) {
```

Comment:
can there be OperationNode?

Reply:
No. Complex queries are not implemented for this API. Refer [code](https://github.com/wso2/identity-api-user/blob/a5b6e3f149d82884bc580f9dca3a64ba54b3ad5f/components/org.wso2.carbon.identity.api.user.application/org.wso2.carbon.identity.rest.api.user.application.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/application/v1/core/ApplicationService.java#L185)